### PR TITLE
ec2 refactor, unit test correction

### PR DESCRIPTION
**Pull request by prekoa**

### CODE REVIEW

File: productinfo/ec2/productinfo_ec2_test.go [hunk 1]
```diff
@@ -6,16 +6,17 @@ import (
...
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/pricing"
 	"github.com/banzaicloud/telescopes/productinfo"
 	"github.com/stretchr/testify/assert"
 )
 
-type DummyPricingSource struct {
+type dummy struct {
```
Comment (lpuskas) [inline productinfo/ec2/productinfo_ec2_test.go]: please use a better name for this and document what it is intended for

File: productinfo/ec2/productinfo_ec2_test.go [hunk 2]
```diff
@@ -143,10 +144,67 @@ func (dps *DummyPricingSource) GetProducts(input *pricing.GetProductsInput) (*pr
...
 }
 
+func (dps *dummy) DescribeAvailabilityZones(input *ec2.DescribeAvailabilityZonesInput) (*ec2.DescribeAvailabilityZonesOutput, error) {
+	if dps.TcId == 10 {
+		return nil, errors.New("could not get information about zones")
+	}
+	return &ec2.DescribeAvailabilityZonesOutput{
+		AvailabilityZones: []*ec2.AvailabilityZone{
+			{
+				State:      aws.String("available"),
```
Comment (lpuskas) [inline productinfo/ec2/productinfo_ec2_test.go]: please check if there are constants in the ec2 library for the strings here

File: productinfo/ec2/productinfo_ec2_test.go [hunk 3]
```diff
@@ -317,42 +404,134 @@ func TestEc2Infoer_GetRegion(t *testing.T) {
...
+		name       string
+		region     string
+		ec2CliMock func(region string) Ec2Describer
+		check      func(data map[string]productinfo.SpotPriceInfo, err error)
 	}{
 		{
-			name: "success - create a new config instance",
-			check: func(config *aws.Config) {
-				assert.NotNil(t, config, "the config should not be nil")
+			name:   "successful - get current sport prices",
```
Comment (lpuskas) [inline productinfo/ec2/productinfo_ec2_test.go]: typo: sport price

File: productinfo/ec2/productinfo_ec2_test.go (hunk 2 above)
Comment (prekoa) [reply productinfo/ec2/productinfo_ec2_test.go]: thx

File: productinfo/ec2/productinfo_ec2_test.go (hunk 3 above)
Comment (prekoa) [reply productinfo/ec2/productinfo_ec2_test.go]: fixed, thx

**Comment by lpuskas** [comment]
Fixes #55

**Pull request merged**